src/Microsoft.ApiDesignGuidelines.Analyzers/UnitTests/CollectionsShouldImplementGenericInterfaceTests.cs

```diff
@@ -12,12 +12,366 @@ public class CollectionsShouldImplementGenericInterfaceTests : DiagnosticAnalyze
     {
         protected override DiagnosticAnalyzer GetBasicDiagnosticAnalyzer()
         {
-            return new BasicCollectionsShouldImplementGenericInterfaceAnalyzer();
+            return new CollectionsShouldImplementGenericInterfaceAnalyzer();
         }
 
         protected override DiagnosticAnalyzer GetCSharpDiagnosticAnalyzer()
         {
-            return new CSharpCollectionsShouldImplementGenericInterfaceAnalyzer();
+            return new CollectionsShouldImplementGenericInterfaceAnalyzer();
+        }
+
+        [Fact]
+        public void Test_WithCollectionBase()
```

Comment:
If the analyzer is implementing only one rule ,The ruleId can be skipped from the TestName and if the same Test contains both VB and CSharp verify steps then the language can be skipped too.